### DISCUSSION TITLE
I want to change option `--skip-drop` to `--enable-drop`

### DISCUSSION THREAD

**Discussion by masak1yu** [sqldef/sqldef#678] [Feature requests]
Thank you for the marvelous tool!
Currently, if used without any options, some diffs will drop the table.
If not careful with this specification, there is a risk of unintended table deletion.
As a request, I would like you to change so that it skips dropping tables without options and only drops tables with options.

```
before
--skip-drop

after
--enable-drop
```

f.y.i
https://github.com/ridgepole/ridgepole/issues/105

Please confirm.

**Comment by k0kubun** [comment]
Sounds good to me. 

Should the option target only tables, or do you want to skip dropping indexes, constraints, columns, and everything else by default too? I wonder whether what you really want is `--enable-drop-table` like ridgepole (disabling only `DROP TABLE` by default) or just `--enable-drop` (disable any `DROP` DDLs by default).

**Comment by masak1yu** [comment]
I think, Skipping all the latter all DDL seems to me more reasonable in production.
Therefore, I think the option is `--enable-drop`.

**Comment by k0kubun** [comment]
OK. Please feel free to file a patch that adds `--enable-drop` to sqldef commands. I'll accept it.

> I think, Skipping all the latter all DDL seems to me more reasonable in production.

Just to ground your design decision, can you provide a more concrete use case? Here's an example use case, which is a counter-argument to your claim (I'm not saying it should be `--enable-drop-table` here, but at least the following text seems more compelling than what you've just said :slightly_smiling_face:):

---

Let's say you have only `users` table in development and CI. You have `users` and `users_backup` in production. When you start using sqldef, you added only `users` table because you can only see that in development and CI. Then, you tried applying the schema to production. You managed to notice that it's going to `DROP TABLE users_backup`, but you felt scared that it could drop such tables. Dropping tables is a fairly rare operation, so you're okay with making it disabled by default.

Now, you found that `users` table has an unnecessary index that used to be in use by a query that no longer exists. To speed up INSERT, you want to drop the index. You removed the index definition from your schema.sql. CI automation executed sqldef, but it wouldn't delete the index if any DROP is disabled by default. Unlike `users_backup`, you're pretty much aware of the `users` table and everything you have in your schema.sql. So when you remove the index, you mean it. There's no circumstance where you delete a line of an index when you don't want to delete the index.

**Comment by masak1yu** [comment]
In the above example, it seems that there is little reason to suppress the drop for index.
(I think the same is true for constraints)
I think that it is better to suppress the drop in consideration of the use in production because the column may contain the user's asset data.
However, I can't think of any use cases where columns are unintentionally dropped.

**Comment by k0kubun** [comment]
I think it's fairly rare that a list of indexes, constraints, or comments differs between a production table and a non-production one, compared to a list of production tables vs a list of non-production ones. I feel like the proposed solution is sacrificing necessary convenience for mitigating risks that don't actually exist in many cases.

**Comment by masak1yu** [comment]
As you said, even if you consider various risks, if you only deal with the parts that can realistically occur, I think that suppressing only the drop of the table will not be a problem.

**Comment by k0kubun** [comment]
:+1: Let's suppress only table drops then.

**Comment by k0kubun** [comment]
This has been implemented :slightly_smiling_face: Closing.
